### ISSUE TITLE
'An items for permanent removing not found'

### ISSUE THREAD

**Issue by q00u** [Amaimersion/remove-files-webpack-plugin#15]
I'm now getting this error:
```
remove-files-plugin@1.2.1:
Skipped, because unsafe removing – "C:\Users\q00u\dev\web-baseline\build"
An items for permanent removing not found
```

My settings are:
```javascript
new RemovePlugin({
  after: {
    test: [{
      folder: 'build',
      method: filepath => {
        return new RegExp(/\.(js|map|html|png|LICENSE)$/, 'm').test(filepath);
       }
    }],
    trash: false
  }
})
```

If I leave `trash` as the default true, the error is: `An items for removing in trash not found`

1.1.3 works as expected, this is a new issue since 1.2.0

**Comment by Amaimersion** [comment]
Hello!

Please enable `logDebug` parameter, run this process again and send all logs from console in this issue.

**Comment by q00u** [comment]
Hrmm... I reinstalled 1.2.1 and added logDebug, and .... it worked? No error message anymore.

I changed nothing else other than reinstalling 1.2.1. I have no explanation.

Closing for now, since everything is working without error.

**Comment by q00u** [comment]
It's back. I added a `before:` section, and the strange "An items for permanent removing not found" has returned. It _doesn't_ happen on the `after:` section.

My webpack entry:
```javascript
new RemovePlugin({
	before: {
		test: [{
			folder: 'build',
			method: filepath => {
				return new RegExp(/\.(LICENSE)$/,'m').test(filepath);
			}
		}],
		trash: false,
		logDebug: true
	},
	after: {
		test: [{
			folder: 'build',
			method: filepath => {
				return new RegExp(/\.(js|map|html|png|LICENSE)$/, 'm').test(filepath);
			}
		}],
		trash: false
	}
})
```

The resulting log:
```
remove-files-plugin@1.2.2:
An items for permanent removing not found

remove-files-plugin@1.2.2:
v4 hook registered – "beforeRun"
v4 hook registered – "watchRun"
v4 hook registered – "afterEmit"
Hook started – "beforeRun"
"build" converted to "C:\Users\q00u\dev\web-baseline\build"
Before formatting:
Root – "C:\Users\q00u\dev\web-baseline"
Path – "C:\Users\q00u\dev\web-baseline\build"
After formatting:
Root – "C:\\\\Users\\\\q00u\\\\dev\\\\web-baseline
Path – "C:\\Users\\q00u\\dev\\web-baseline\\build"
Save – "true"
Test not passed – "C:\Users\q00u\dev\web-baseline\build\Demo_2.3.0.zip"
Directories before cropping – 
Files before cropping – 
Directories after cropping – 
Files after cropping – 
An items for permanent removing not found
Hook ended – "beforeRun"
```

This is in version 1.2.2

**Comment by Amaimersion** [comment]
Hello @q00u 

_before_ build you have only `Demo_2.3.0.zip` file in `build` folder. You specified test `return new RegExp(/\.(LICENSE)$/,'m').test(filepath);`. However, this test will results in true only for `*.LICENSE` files. So, you don't have any files or folders _before_ build that will pass this test. And that's why plugin logs `An items for permanent removing not found`. See [this](https://regex101.com/r/eTuKk8/1).

I see you also have  `*.LICENSE` in _after_ section. Be careful, because it will not remove `LICENSE` file, it will remove `*.LICENSE` files (files that have `.LICENSE` extension).

If you want to remove `*.zip` files before build, you can achieve this with `return new RegExp(/\.(zip)$/,'m').test(filepath);`.

If you also want to remove `LICENSE` file, you can achieve this with 
```javascript
before: {
    include: [
        'LICENSE'
    ],
    ...
},
```

**Comment by q00u** [comment]
I see! So it's not unable to delete files like before, but rather it isn't finding files to delete.

(It turned out that the *.LICENSE files were being generated by the Terser during compilation, so they weren't there during Before).

Thanks!
